packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1977,6 +1985,59 @@ export class ContainerRuntime
 		this.removeAllListeners();
 	}
 
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#IContainerRuntimeBase.isSnapshotFetchRequired}
+	 */
+	public isSnapshotFetchRequired(pathParts: string[]): boolean {
+		validatePrecondition(
+			this.snapshotWithContents !== undefined,
+			"Use this api in case we fetched the complete snapshot along with the blob contents",
+		);
+		// Find the snapshotTree inside the snapshot based on the path as given in the request.
+		const hasIsolatedChannels = rootHasIsolatedChannels(this.metadata);
+		const snapshotTreeForPath = this.getSnapshotTreeForPath(
+			this.snapshotWithContents.snapshotTree,
+			pathParts,
+			hasIsolatedChannels,
+		);
+		if (snapshotTreeForPath === undefined) {
+			throw new UsageError("snapshotTree not found for given path", {
+				details: JSON.stringify({ pathPartsLength: pathParts.length }),
+			});
+		}
+		return this.evaluateSnapshotTreeForMissingBlobs(
```

Comment:
Why data store context needs to call into here to make this evaluation? I.e. what information it does not have to do the same evaluation locally? It has access to a snapshot, right? Maybe it does not have access to "extended" snapshot to evaluate if blob is there - if so, I'd rather pass that info down, then do this call up the stack.

Reply:
Moved this to datastorecontext.